lib/auth/state.go

```diff
@@ -29,16 +29,33 @@ import (
 	"github.com/gravitational/trace"
 )
 
+// stateBackend implements abstraction over local or remote storage backend methods
+// required for Identity/State storage.
+// As in backend.Backend, Item keys are assumed to be valid UTF8, which may be enforced by the
+// various Backend implementations.
+type stateBackend interface {
+	// Create creates item if it does not exist
+	Create(ctx context.Context, i backend.Item) (*backend.Lease, error)
+	// Put puts value into backend (creates if it does not
+	// exists, updates it otherwise)
+	Put(ctx context.Context, i backend.Item) (*backend.Lease, error)
+	// Get returns a single item or not found error
+	Get(ctx context.Context, key []byte) (*backend.Item, error)
+}
+
 // ProcessStorage is a backend for local process state,
 // it helps to manage rotation for certificate authorities
 // and keeps local process credentials - x509 and SSH certs and keys.
 type ProcessStorage struct {
-	backend.Backend
+	BackendStorage backend.Backend
+	stateStorage   stateBackend
```

Comment:
The fields here deserve some comment, especially in the light of `stateStorage` not being closed in `Close`.